fair_mango/metrics/superset.py

```
@@ -75,112 +87,120 @@ def __init__(
-        ...         'real_churn': [1,1,0,0,1],
-        ...         'pred_churn': [0,1,0,0,1]
-        ... })
-        >>> super_set_fairness_metrics = SupersetFairnessMetrics(
-        ...     metric=DemographicParityDifference,
-        ...     data=df,
-        ...     sensitive=['gender', 'race'],
-        ...     real_target=['real_churn'],
-        ...     predicted_target=['pred_churn'],
-        ... )
-        >>> result = super_set_fairness_metrics.rank()
-        >>> result
-        [
-            {
-                'sensitive': ('gender',),
-                'result': {
-                    'real_churn': {
-                        ('male',): 0.16666666666666663,
-                        ('female',): -0.16666666666666663
-                    }
-                }
-            },
-            {
-                'sensitive': ('race',),
-                'result': {
-                    'real_churn': {
-                        ('white',): 0.16666666666666663,
-                        ('black',): -0.16666666666666663
-                    }
-                }
-            },
-            {
-                'sensitive': ('gender', 'race'),
-                'result': {
-                    'real_churn': {
-                        ('male', 'white'): 0.5,
-                        ('female', 'white'): -0.25,
-                        ('male', 'black'): -0.25
-                    }
-                }
-            }
-        ]
+        for pair in self.pairs:
+            dataset = Dataset(
+                self.df,
+                list(pair),
+                self.real_target,
+                self.predicted_target,
+                self.positive_target,
+            )
+
+            rankings = {}
+
+            for metric_name, metric_class in self._dataset_metrics.items():
+                try:
+                    metric = metric_class(dataset)
+                    rankings[metric_name] = metric.rank()
+                except Exception as e:
```

1. do not catch general exceptions.
2. either let it crash, or reraise the exception
3. never ignore general exceptions (by using the continue statement or by doing nothing)
4. don't use print, use logging instead
5. if you wanna detect a specific failure, you can raise a custom exception in the try: block and catch it with the except: block

---

still not resolved. I can see two general exception catching `except Exception as e`

---

up, you need to resolve points 1, 2, and 5

---

resolved